docs/capacity-planning.rst

```diff
@@ -0,0 +1,326 @@
+
+**Data Schema**: KSQL handles mapping serialized Kafka records to columns in a stream or table’s schema. In general, more complex schemas with a higher ratio of columns to bytes of data require more CPU to process.
+
+**Number of Partitions**: Kafka Streams creates one RocksDB state store instance for aggregations and joins for every topic partition processed by a given KSQL server. Each RocksDB state store instance has a memory overhead of 50MB for its cache plus the data actually stored.
+
+**Key Space**: For aggregations and joins, Kafka Streams/RocksDB will try and keep the working set of a state store in memory to avoid I/O operations. If there are many keys then this will require more memory. It also makes reads and writes to the state store more expensive. Note that the size of the data in a state store is not limited by memory (RAM) but only by available disk space on a KSQL server.
+
+Recommendations and Best Practices
+==================================
+
+Interactive KSQL Servers vs Non-Interactive (“Headless”) Servers 
+----------------------------------------------------------------
+
+By default, KSQL servers are configured for interactive use, which means you can use the KSQL CLI to interact with a KSQL cluster in order to, for example, execute new queries. Interactive KSQL usage allows for easy and quick iterative development and testing of your KSQL queries via the KSQL CLI.
+
+You can also configure the servers for headless, non-interactive operation, where servers collaboratively run only a predefined list of queries [link to KSQL server configuration section]. The result is essentially an elastic, fault-tolerant, and distributed stream processing application that communicates to the outside world by reading from and writing to Kafka topics.  Sizing, deploying, and managing in this scenario is similar to a Kafka Streams application [link to KStreams docs] . We also recommend to integrate such KSQL deployments with your company's CI/CD pipeline to, for example, version-control the .sql file.
+
+How to choose between these two configuration types?
+
+    - For production deployments we recommend headless, non-interactive KSQL clusters. This configuration provides the best isolation and, unlike interactive KSQL clusters, minimizes the likelihood of operator error and human mistakes.
+    - For exploring and experimenting with your data we recommend interactive KSQL clusters. It allows to quickly come up with sets of desired queries for your use case that together function as a streaming “application” to produce meaningful results. Then, run this “application” with headless, non-interactive KSQL clusters in production.
```

Comment:
`For exploring and experimenting with your data, interactive KSQL clusters are recommended.`